src/components/codecSupportHelper.ts

```diff
@@ -102,22 +102,28 @@ export function getMaxBitrateSupport(): number {
 /**
  * Get the max supported video width the active Cast device supports.
  *
- * @param deviceId - Cast device id.
  * @returns Max supported width.
  */
-export function getMaxWidthSupport(deviceId: number): number {
-    switch (deviceId) {
-        case deviceIds.ULTRA:
-        case deviceIds.CCGTV:
-            return 3840;
-        case deviceIds.GEN1AND2:
-        case deviceIds.GEN3:
-            return 1920;
-        case deviceIds.NESTHUBANDMAX:
-            return 1280;
-    }
-
-    return 0;
+export function getMaxWidthSupport(): number {
+    // with HLS, it will produce a manifest error if we
+    // send any stream larger than the screen size...
+    return window.innerWidth;
```

Comment:
This causes all my videos to be transcodeded even though they can be direct played (setups: CCUltra/4K-TV and CCGen3/FHD-TV). window.innerWidth was 1280 when I tried with gen 2/3 chromecast and a FHD TV. 1920 preferred MaxWidth.

Maybe only use window.innerWidth when HLS stream?

Reply:
@ownbee what should the value be if it's not an HLS stream (instead of `window.innerWidth`) ?

Reply:
How does [this](https://github.com/hawken93/jellyfin-chromecast/pull/228/commits/9e668e1e916a5bc123737344d8faa54fb98d03df) look?

Reply:
@tnyeanderson  I think they should be clamped depending on CC-type as it was before. For direct play I think the CC/TV is able to automatically scale if the resolution is too big for the screen (it has worked before). For HLS I have no clue how to solve it.

I fail to see how `h264` coding has anything to do with MaxWidthSupport. If I play a 1920x1080 video with `h264` coding on my FHD TV (1920x1080) do I get the `window.innerWidth` constraint? If so, I would say this is not that great since 95% of my movies are in that format and would be transcoded to lower resolution and put a lot of load on my server.

Problem is, I think, is that the jellyfin server is the one that decides if transcoding is needed and how to stream depending on the constraints chromecast-app reports. Which means this logic suits best to be in the jellyfin server or if we could find something better than `window.innerWidth` or if we could tell it "1920x1080 is okay but if you are going to transcode for other reasons, use `window.innerWidth`".

I'll try find time to test if this works and how `window.innerWidth` behaves this weekend but I can't make a promise.

Thanks for your effort trying to get this in! Really appreciated!